### DISCUSSION TITLE
How to access additional properties of data from Autocomplete on submit?

### DISCUSSION THREAD

**Discussion by RedFour** [mantinedev/mantine#2248] [Q&A]
I'm using Autocomplete with custom data and Mantine useForm.

`{
    value: string;
    id: string;
}`

When user submits the form, I would like to access the underlining id. 

`{...form.getInputProps('searchValue')}`

`getInputProps` only gets me the value of the Autocomplete. How do I get the associated `id` when the user selects one of the existing data?

**Comment by rtivital** [comment]
`data.find(item => item.value === form.selectedAutocompleteItem)`

**Reply by RedFour** [reply]
My data can have multiple items where the value are the same but other properties are not. I plan to use itemComponent to show the differences between those items. Value, which is the name of the item, is still something I want Autocomplete to search with.

That's why I want the exact id associated with item selected by the user, instead of doing a search within my data.

Is there a way to have value be the id, but still have Autocomplete perform its search with another field?

**Reply by rtivital** [reply]
Items with the same `value` property are not supported – it will break the component. Value must always be unique. You can add any other properties to the component that can be duplicated, for example `id`. You can customize items filtering with `filter` prop.
